src/libraries/System.Security.Cryptography/tests/HashAlgorithmNamesTests.cs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Security.Cryptography.Tests
+{
+    public static class HashAlgorithmNamesTests
+    {
+        [Theory]
+        [MemberData(nameof(ValidInputs))]
+        public static void ToUpper_UnchangedValidInput(string hashAlgorithmName)
+        {
+            string actual = HashAlgorithmNames.ToUpper(hashAlgorithmName);
+            Assert.Equal(hashAlgorithmName, actual);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidInputs))]
+        public static void ToUpper_FixesLoweredValidInput(string hashAlgorithmName)
+        {
+            string actual = HashAlgorithmNames.ToUpper(hashAlgorithmName.ToLowerInvariant());
+            Assert.Equal(hashAlgorithmName, actual);
+        }
+
+        [Fact]
+        public static void ToUpper_UnchangedFromInvalidInput()
+        {
+            string unsupported = UnsupportedHashAlgorithm.ID.ToLowerInvariant();
+            string actual = HashAlgorithmNames.ToUpper(unsupported);
+            Assert.Equal(unsupported, actual);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidInputs))]
+        public static void ToAlgorithmName_ValidInput(string hashAlgorithmName)
+        {
+            bool hashSupported = HashProviderDispenser.HashSupported(hashAlgorithmName);
+
+            if (hashSupported)
+            {
+                using (HashAlgorithm hashAlgorithm = CreateHashAlgorithm(hashAlgorithmName);
```

Comment:
missing `)`

```suggestion
                using (HashAlgorithm hashAlgorithm = CreateHashAlgorithm(hashAlgorithmName));
```

Reply:
Not sure how this happened, many apologies for the bad rebase...